### DISCUSSION TITLE
Getting error trying to deploy app to vercel

### DISCUSSION THREAD

**Discussion by Emabliss** [prisma/prisma#10478] [Q&A] [answered]
There's this basic full-stack blog app I built using Nextjs, Prisma, and PostgreSQL that I've been having trouble deploying to Vercel. It says I should install prisma/client as a peer dependency of next-auth. I'm kind of confused here because I already installed it and it's in my package.json file. Help please!

**Comment by austincrim** [comment, marked as answer]
Hey @Emabliss 👋,

Would you mind sharing the full error message and the relevant parts from your `package.json`?

**Reply by Emabliss** [reply]
Here's the complete log:

Cloning github.com/Emabliss/emanuel-ameh-blogr-nextjs-prisma (Branch: main, Commit: a44fe93)
--
21:18:52.493 | Cloning completed: 766.776ms
21:18:52.546 | Analyzing source code...
21:18:53.563 | Installing build runtime...
21:18:55.979 | Build runtime installed: 2.416s
21:18:58.340 | No Build Cache available
21:18:59.492 | Installing dependencies...
21:18:59.805 | yarn install v1.22.17
21:18:59.889 | warning package-lock.json found. Your project contains lock files generated by tools other than Yarn. It is advised not to mix package managers in order to avoid resolution inconsistencies caused by unsynchronized lock files. To clear this warning, remove package-lock.json.
21:18:59.893 | [1/4] Resolving packages...
21:19:03.218 | [2/4] Fetching packages...
21:19:22.293 | [3/4] Linking dependencies...
21:19:22.303 | warning "next-auth > @next-auth/prisma-legacy-adapter@0.1.2" has incorrect peer dependency "@prisma/client@^2.16.1".
21:19:29.733 | [4/4] Building fresh packages...
21:19:36.890 | success Saved lockfile.
21:19:36.895 | $ prisma generate \|\| true
21:19:38.464 | Prisma schema loaded from prisma/schema.prisma
21:19:40.064 |  
21:19:40.064 | ✔ Generated Prisma Client (3.5.0) to ./node_modules/@prisma/client in 149ms
21:19:40.065 | You can now start using Prisma Client in your code. Reference: https://pris.ly/d/client
21:19:40.065 | ```
21:19:40.065 | const prisma = new PrismaClient()
21:19:40.065 | ```
21:19:40.142 | Done in 40.34s.
21:19:40.182 | Detected Next.js version: 9.5.5
21:19:40.190 | Running "yarn run build"
21:19:40.572 | yarn run v1.22.17
21:19:40.610 | $ next build
21:19:41.424 | Browserslist: caniuse-lite is outdated. Please run:
21:19:41.425 | npx browserslist@latest --update-db
21:19:41.529 | info  - Creating an optimized production build...
21:19:41.547 | Attention: Next.js now collects completely anonymous telemetry regarding usage.
21:19:41.547 | This information is used to shape Next.js' roadmap and prioritize features.
21:19:41.547 | You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
21:19:41.547 | https://nextjs.org/telemetry
21:19:41.547 |  
21:19:45.903 | Failed to compile.
21:19:45.903 |  
21:19:45.903 | ./lib/prisma.ts:8:15
21:19:45.904 | Type error: Property 'prisma' does not exist on type 'Global & typeof globalThis'.
21:19:45.904 |  
21:19:45.904 | [0m [90m  6 \| [39m  prisma [33m=[39m [36mnew[39m [33mPrismaClient[39m()[33m;[39m[0m
21:19:45.904 | [0m [90m  7 \| [39m} [36melse[39m {[0m
21:19:45.904 | [0m[31m[1m>[22m[39m[90m  8 \| [39m  [36mif[39m ([33m![39mglobal[33m.[39mprisma) {[0m
21:19:45.904 | [0m [90m    \| [39m              [31m[1m^[22m[39m[0m
21:19:45.905 | [0m [90m  9 \| [39m    global[33m.[39mprisma [33m=[39m [36mnew[39m [33mPrismaClient[39m()[33m;[39m[0m
21:19:45.905 | [0m [90m 10 \| [39m  }[0m
21:19:45.929 | error Command failed with exit code 1.
21:19:45.929 | info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
21:19:45.950 | Error: Command "yarn run build" exited with 1


</body>
</html>

**Reply by Emabliss** [reply]
And package.json:
"dependencies": {
    "@next-auth/prisma-adapter": "^0.5.2-next.19",
    "@prisma/client": "^3.5.0",
    "next": "9.5.5",
    "next-auth": "^3.29.0",
    "react": "16.14.0",
    "react-dom": "16.14.0",
    "react-markdown": "5.0.2"
  },
  "devDependencies": {
    "@types/next-auth": "3.1.14",
    "@types/node": "14.14.6",
    "@types/react": "16.9.56",
    "prisma": "^3.5.0",
    "typescript": "^4.5.2"
  }
Thanks in advance

**Reply by austincrim** [reply]
Does `next build` pass locally?

**Reply by Emabliss** [reply]
No, it doesn't. It instead points me to a file:
./lib/prisma.ts:8:15
Type error: Property 'prisma' does not exist on type 'Global & 
typeof globalThis'.

   6 |   prisma = new PrismaClient();
   7 | } else {
>  8 |   if (!global.prisma) {
     |               ^
   9 |     global.prisma = new PrismaClient();
  10 |   }
  11 |   prisma = global.prisma;

**Reply by austincrim** [reply]
Are you using this snippet from [our docs](https://www.prisma.io/docs/support/help-articles/nextjs-prisma-client-dev-practices#solution)?

```ts
import { PrismaClient } from '@prisma/client'

declare global {
  var prisma: PrismaClient | undefined
}

export const prisma =
  global.prisma ||
  new PrismaClient({
    log: ['query'],
  })

if (process.env.NODE_ENV !== 'production') global.prisma = prisma
```

I think that `declare global` block is especially relevant here.

**Reply by Emabliss** [reply]
That worked perfectly well, thanks. But just one last question: I can delete and update posts in development mode but I can't when it's live. Here are the functions:
```ts
async function publishPost(id: number): Promise<void> {
  await fetch(`http://localhost:3000/api/publish/${id}`, {
    method: 'PUT',
  });
  await Router.push('/');
}

async function deletePost(id: number): Promise<void> {
  await fetch(`http://localhost:3000/api/post/${id}`, {
    method: 'DELETE',
  });
  Router.push('/');
}
```
I wasn't really sure of my endpoints so I changed them to:
` http://emanuel-ameh-blogr-nextjs-prisma.vercel.app//publish/${id}` and ` http://emanuel-ameh-blogr-nextjs-prisma.vercel.app/api/post/${id}` respectively and it still didn't work.

**Reply by austincrim** [reply]
You will definitely need to change the URL you are fetching once you deploy. `localhost:3000` is just the dev server when you are running locally.

If you are using Next.js and API routes, you can remove the hostname entirely and just fetch `/api/publish/${id}`.

If you are still hitting errors, let me know what the error messages are.
Thanks!

**Reply by Emabliss** [reply]
It works well. Thanks!

**Reply by Emabliss** [reply]
Hi, I'm having a problem here. I'm building an app using nextjs and nextauth for authentication. I can login and logout, but the weird thing is in my Header component, session is defined, session.user too is defined; but when I console log session  in other components, I get a result in my browser console but it's undefined in my terminal and session.user returns an error. Please what do you think might be the cause?

**Reply by austincrim** [reply]
I'm not sure, a good place to ask for further help would be the [Next.js Discord](https://discord.com/invite/nextjs)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
